servicetalk-opentelemetry/src/main/java/io/servicetalk/opentelemetry/OpenTelemetryHttpServerFilter.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry;
+
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.http.api.HttpExecutionStrategy;
+import io.servicetalk.http.api.HttpHeaders;
+import io.servicetalk.http.api.HttpRequestMetaData;
+import io.servicetalk.http.api.HttpServiceContext;
+import io.servicetalk.http.api.StreamingHttpRequest;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.api.StreamingHttpResponseFactory;
+import io.servicetalk.http.api.StreamingHttpService;
+import io.servicetalk.http.api.StreamingHttpServiceFilter;
+import io.servicetalk.http.api.StreamingHttpServiceFilterFactory;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+import java.util.function.UnaryOperator;
+
+import static io.servicetalk.http.api.HttpExecutionStrategies.offloadNone;
+
+/**
+ * A {@link StreamingHttpService} that supports open telemetry.
+ * <p>
+ * Append this filter before others that are expected to see {@link Scope} for this request/response. Filters
+ * appended after this filter that use operators with the <strong>after*</strong> prefix on
+ * {@link StreamingHttpService#handle(HttpServiceContext, StreamingHttpRequest, StreamingHttpResponseFactory)
+ * response meta data} or the {@link StreamingHttpResponse#transformMessageBody(UnaryOperator) response message body}
+ * (e.g. {@link Publisher#afterFinally(Runnable)}) will execute after this filter invokes {@link Scope#close()} and
+ * therefore will not see the {@link Span} for the current request/response.
+ */
+public class OpenTelemetryHttpServerFilter implements StreamingHttpServiceFilterFactory {
+    private static final TextMapGetter<HttpHeaders> getter = new HeadersPropagatorGetter();
+    private static final TextMapSetter<HttpHeaders> setter = new HeadersPropagatorSetter();
+
+    private final Tracer tracer;
+    private final ContextPropagators propagators;
+
+    /**
+     * Create a new instance.
+     *
+     * @param openTelemetry the {@link OpenTelemetry}.
+     */
+    public OpenTelemetryHttpServerFilter(final OpenTelemetry openTelemetry) {
+        this.tracer = openTelemetry.getTracer("io.servicetalk");
```

Comment:
Should we make this constant public and document in javadoc that this is what we use?

Reply:
wdyt?